Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSelector.cpp

```diff
@@ -56,6 +62,21 @@ ActsExamples::ProcessCode ActsExamples::ParticleSelector::execute(
   // prepare input/ output types
   const SimParticleContainer& inputParticles = m_inputParticles(ctx);
 
+  // Optionally construct a particles->measurement map
+  std::optional<boost::container::flat_multimap<ActsFatras::Barcode, Index>>
+      partMeasMap;
+  if (m_inputMeasPartMap.isInitialized()) {
+    const auto& measPartMap = m_inputMeasPartMap(ctx);
+    using InvPair = std::pair<ActsFatras::Barcode, Index>;
+    std::vector<InvPair> v(measPartMap.size());
+    std::ranges::transform(measPartMap, v.begin(), [](const auto& p) {
+      return InvPair{p.second, p.first};
+    });
+    std::ranges::sort(v, std::less{}, &InvPair::first);
+    partMeasMap.emplace(boost::container::ordered_range_t{}, v.begin(),
+                        v.end());
+  }
```

Comment:
I thought this inverted map is somewhere generated already and put on the whiteboard?

Also, isn't there a utility to do this kind of transformation? I can also try to look it up